contracts/icp/proxy-contract/src/mutate.rs

```diff
@@ -0,0 +1,304 @@
+                            format!("Transfer failed: {:?}", e)
+                        })?;
+            }
+            ICProposalAction::SetNumApprovals { num_approvals } => {
+                PROXY_CONTRACT.with(|contract| {
+                    let mut contract = contract.borrow_mut();
+                    contract.num_approvals = num_approvals;
+                });
+            }
+            ICProposalAction::SetActiveProposalsLimit {
+                active_proposals_limit,
+            } => {
+                PROXY_CONTRACT.with(|contract| {
+                    let mut contract = contract.borrow_mut();
+                    contract.active_proposals_limit = active_proposals_limit;
+                });
+            }
+            ICProposalAction::SetContextValue { key, value } => {
+                PROXY_CONTRACT.with(|contract| {
+                    let mut contract = contract.borrow_mut();
+                    contract.context_storage.insert(key.clone(), value.clone());
+                });
+            }
+        }
+    }
+
+    remove_proposal(proposal_id.clone());
+    Ok(())
+}
+
+fn internal_create_proposal(
+    proposal: ICProposal,
+    num_proposals: u32,
+) -> Result<Option<ICProposalWithApprovals>, String> {
+    if proposal.actions.is_empty() {
+        return Err("proposal cannot have empty actions".to_string());
+    }
+
+    PROXY_CONTRACT.with(|contract| {
+        let mut contract = contract.borrow_mut();
+
+        // Check proposal limit
+        if num_proposals >= contract.active_proposals_limit {
+            return Err(
+                "Account has too many active proposals. Confirm or delete some.".to_string(),
+            );
+        }
+
+        // Validate proposal actions
+        for action in &proposal.actions {
+            validate_proposal_action(action)?;
+        }
+
+        // Store proposal
+        let proposal_id = proposal.id.clone();
+        contract
+            .proposals
+            .insert(proposal_id.clone(), proposal.clone());
+
+        // Initialize approvals set with author's approval
+        let mut approvals = HashSet::new();
+        approvals.insert(proposal.author_id.clone());
+        contract.approvals.insert(proposal_id.clone(), approvals);
+
+        // Update proposal count
+        let author_id = proposal.author_id;
+        *contract.num_proposals_pk.entry(author_id).or_insert(0) += 1;
+
+        build_proposal_response(&*contract, proposal_id)
+    })
+}
+
+fn validate_proposal_action(action: &ICProposalAction) -> Result<(), String> {
+    match action {
+        ICProposalAction::ExternalFunctionCall {
+            receiver_id: _,
+            method_name,
+            args,
+            deposit: _,
+        } => {
+            if method_name.is_empty() {
+                return Err("method name cannot be empty".to_string());
+            }
+            if args.is_empty() {
+                return Err("args cannot be empty".to_string());
+            }
```

Comment:
the ICP requirement is to encode the arguments for the call. They can be empty but currently, I can't think of a call that wont be needing arguments.

Reply:
since this is an external function call, we cannot preemptively know if the function will have any arguments, can you test introducing a function with no arguments and see if calling it works with no args.